internal/ignition/ignition.go

```diff
@@ -1286,55 +1287,168 @@ func getBootReporterFileContent() (string, error) {
+		return "", err
 	}
+	content, err := io.ReadAll(respBody)
+	if err != nil {
+		return "", err
+	}
+	return string(content), nil
+}
 
-	for i := range hosts {
-		host := hosts[i]
-		errGroup.Go(func() error {
-			config, err := parseIgnitionFile(filepath.Join(workDir, baseFile))
-			if err != nil {
-				return err
+func machineConfilePoolExists(manifestFname, content, poolName string) (bool, error) {
+	var (
+		manifest struct {
+			Kind     string
+			Metadata *struct {
+				Name string
 			}
-			pullSecretToken, err := clusterPkg.AgentToken(g.cluster, authType)
-			if err != nil {
-				return err
-			}
-			contents := fmt.Sprintf(assistedBootReporterunitTemplate, strings.TrimSpace(serviceBaseURL), pullSecretToken, host.ClusterID, host.ID, 5, 60)
-			setUnitInIgnition(config, contents, "assisted-boot-reporter.service", true)
+		}
+		err error
+	)
+	ext := filepath.Ext(manifestFname)
+	switch ext {
+	case ".yml", ".yaml":
+		err = yaml.Unmarshal([]byte(content), &manifest)
+	case ".json":
+		err = json.Unmarshal([]byte(content), &manifest)
+	default:
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+	return manifest.Kind == "MachineConfigPool" && manifest.Metadata != nil && manifest.Metadata.Name == poolName, nil
+}
 
-			hostname, err := hostutil.GetCurrentHostName(host)
-			if err != nil {
-				return errors.Wrapf(err, "failed to get hostname for host %s", host.ID)
-			}
+func (g *installerGenerator) clusterHasMCP(poolName string, clusterId *strfmt.UUID) (bool, error) {
+	var err error
+	ctx := context.Background()
+	manifestList, err := manifests.GetClusterManifests(ctx, clusterId, g.s3Client)
+	if err != nil {
+		return false, err
+	}
+	for _, manifest := range manifestList {
+		content, err := g.getManifestContent(ctx, manifest)
+		if err != nil {
+			return false, err
+		}
+		exists, err := machineConfilePoolExists(manifest, content, poolName)
+		if err != nil {
+			return false, err
+		}
+		if exists {
+			return true, nil
+		}
+	}
+	return false, nil
+}
 
-			setFileInIgnition(config, "/etc/hostname", fmt.Sprintf("data:,%s", hostname), false, 420, true)
+func (g *installerGenerator) updatePointerIgnitionMCP(poolName string, ignitionStr string) (string, error) {
+	config, err := ParseToLatest([]byte(ignitionStr))
+	if err != nil {
+		return "", err
+	}
+	for i := range config.Ignition.Config.Merge {
+		r := &config.Ignition.Config.Merge[i]
+		if r.Source != nil {
+			r.Source = swag.String(strings.Replace(swag.StringValue(r.Source), "config/worker", "config/"+poolName, 1))
+		}
+	}
+	b, err := json.Marshal(config)
+	if err != nil {
+		return "", err
+	}
+	return string(b), nil
+}
 
-			setFileInIgnition(config, "/usr/local/bin/assisted-boot-reporter.sh", fmt.Sprintf("data:text/plain;charset=utf-8;base64,%s", bootReporter), false, 0700, true)
+func (g *installerGenerator) modifyPointerIgnitionMCP(poolName string, ignitionStr string, clusterId *strfmt.UUID) (string, error) {
```

Comment:
This logic will run only on day1 flow right?We should be sure it will be disabled in hypershift and day2 flow